In [1]:
# Import libraries
import pandas as pd

In [2]:
# Load csv file
matches = pd.read_csv("api/ipl_data/matches.csv")

In [3]:
# Verifying csv data
matches.head()

id  season       city        date                        team1  \
0   1    2017  Hyderabad  2017-04-05          Sunrisers Hyderabad   
1   2    2017       Pune  2017-04-06               Mumbai Indians   
2   3    2017     Rajkot  2017-04-07                Gujarat Lions   
3   4    2017     Indore  2017-04-08       Rising Pune Supergiant   
4   5    2017  Bangalore  2017-04-08  Royal Challengers Bangalore   

                         team2                  toss_winner toss_decision  \
0  Royal Challengers Bangalore  Royal Challengers Bangalore         field   
1       Rising Pune Supergiant       Rising Pune Supergiant         field   
2        Kolkata Knight Riders        Kolkata Knight Riders         field   
3              Kings XI Punjab              Kings XI Punjab         field   
4             Delhi Daredevils  Royal Challengers Bangalore           bat   

   result  dl_applied                       winner  win_by_runs  \
0  normal           0          Sunrisers Hyderabad           35   
1  normal           0       Rising Pune Supergiant            0   
2  normal           0        Kolkata Knight Riders            0   
3  normal           0              Kings XI Punjab            0   
4  normal           0  Royal Challengers Bangalore           15   

   win_by_wickets player_of_match                                      venue  \
0               0    Yuvraj Singh  Rajiv Gandhi International Stadium, Uppal   
1               7       SPD Smith    Maharashtra Cricket Association Stadium   
2              10         CA Lynn     Saurashtra Cricket Association Stadium   
3               6      GJ Maxwell                     Holkar Cricket Stadium   
4               0       KM Jadhav                      M Chinnaswamy Stadium   

          umpire1        umpire2  umpire3  
0     AY Dandekar       NJ Llong      NaN  
1  A Nand Kishore         S Ravi      NaN  
2     Nitin Menon      CK Nandan      NaN  
3    AK Chaudhary  C Shamshuddin      NaN  
4             NaN            NaN      NaN

In [5]:
# Size
matches.shape

(636, 18)

### Top 4 teams in terms of wins

In [13]:
# Winning for all the season specific data
matches.groupby('season')['winner'].value_counts().sort_values(ascending=False)[:4]

season  winner               
2008    Rajasthan Royals         13
2013    Mumbai Indians           13
2014    Kings XI Punjab          12
2012    Kolkata Knight Riders    12
Name: winner, dtype: int64

In [14]:
# For all the season total winning
matches.groupby('winner')['winner'].count().sort_values(ascending=False)[:4]

winner
Mumbai Indians                 92
Chennai Super Kings            79
Kolkata Knight Riders          77
Royal Challengers Bangalore    73
Name: winner, dtype: int64

In [16]:
# For specific season
matches.loc[matches['season'] == 2008].groupby('winner')['winner'].count().sort_values(ascending=False)[:4]

winner
Rajasthan Royals       13
Kings XI Punjab        10
Chennai Super Kings     9
Mumbai Indians          7
Name: winner, dtype: int64

### Which team won the most number of tosses in the season

In [18]:
# For all the season total winning
matches.groupby('toss_winner')['toss_winner'].count().sort_values(ascending=False)[:1]

toss_winner
Mumbai Indians    85
Name: toss_winner, dtype: int64

In [20]:
# For specific season
matches.loc[matches['season'] == 2008].groupby('toss_winner')['toss_winner'].count().sort_values(ascending=False)[:1]

toss_winner
Rajasthan Royals    11
Name: toss_winner, dtype: int64

### Which player won the maximum number of Player of the Match awards in the whole season

In [22]:
# For all the season
matches.groupby('player_of_match')['player_of_match'].count().sort_values(ascending=False)[:1]

player_of_match
CH Gayle    18
Name: player_of_match, dtype: int64

### Which team won max matches in the whole season

In [23]:
# For all the season total winning
matches.groupby('winner')['winner'].count().sort_values(ascending=False)[:1]

winner
Mumbai Indians    92
Name: winner, dtype: int64

### Which location has the most number of wins for the top team

In [27]:
matches.groupby('venue')['winner'].value_counts().sort_values(ascending=False)[:1]

venue         winner               
Eden Gardens  Kolkata Knight Riders    37
Name: winner, dtype: int64

### Which % of teams decided to bat when they won the toss

In [30]:
matches.toss_decision.value_counts(normalize=True).mul(100).round(2).astype(str) + "%"

field    57.08%
bat      42.92%
Name: toss_decision, dtype: object

### Which location hosted most number of matches 

In [33]:
matches.groupby('venue')['venue'].count().sort_values(ascending=False)[:1]

venue
M Chinnaswamy Stadium    66
Name: venue, dtype: int64

### Which team won by the highest margin of runs  for the season

In [37]:
matches[['win_by_runs', 'winner']].sort_values(by='win_by_runs', ascending=False)[:2]

win_by_runs                       winner
43           146               Mumbai Indians
619          144  Royal Challengers Bangalore

In [40]:
matches[['win_by_runs', 'winner']].sort_values(by='win_by_runs', ascending=False)['winner'][:1]

43    Mumbai Indians
Name: winner, dtype: object

### Which team won by the highest number of wickets for the season

In [45]:
data = matches.loc[matches['season'] == 2011]
data[['win_by_wickets', 'winner']].sort_values(by='win_by_wickets', ascending=False)[:5]

win_by_wickets                       winner
298              10             Rajasthan Royals
241               9               Mumbai Indians
287               9  Royal Challengers Bangalore
257               9  Royal Challengers Bangalore
245               9        Kolkata Knight Riders

### How many times has a team won the toss and the match

In [48]:
data = matches.loc[matches['toss_winner'].str.lower() == matches['winner'].str.lower()]

In [49]:
data.shape

(325, 18)

325 times has a team won the toss and the match

### Which Batsman gave away the most number of runs in a match for the selected season

In [55]:
# Load csv file
deliveries = pd.read_csv("api/ipl_data/deliveries.csv")

In [56]:
deliveries.head(5)

match_id  inning         batting_team                 bowling_team  over  \
0         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
1         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
2         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
3         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   
4         1       1  Sunrisers Hyderabad  Royal Challengers Bangalore     1   

   ball    batsman non_striker    bowler  is_super_over  ...  bye_runs  \
0     1  DA Warner    S Dhawan  TS Mills              0  ...         0   
1     2  DA Warner    S Dhawan  TS Mills              0  ...         0   
2     3  DA Warner    S Dhawan  TS Mills              0  ...         0   
3     4  DA Warner    S Dhawan  TS Mills              0  ...         0   
4     5  DA Warner    S Dhawan  TS Mills              0  ...         0   

   legbye_runs  noball_runs  penalty_runs  batsman_runs  extra_runs  \
0            0            0             0             0           0   
1            0            0             0             0           0   
2            0            0             0             4           0   
3            0            0             0             0           0   
4            0            0             0             0           2   

   total_runs  player_dismissed dismissal_kind fielder  
0           0               NaN            NaN     NaN  
1           0               NaN            NaN     NaN  
2           4               NaN            NaN     NaN  
3           0               NaN            NaN     NaN  
4           2               NaN            NaN     NaN  

[5 rows x 21 columns]

In [57]:
deliveries.shape

(150460, 21)

In [62]:
# For all the season
deliveries.groupby(['batsman'])['batsman_runs'].sum().reset_index().sort_values(by='batsman_runs', ascending=False)[:1]

batsman  batsman_runs
374  SK Raina          4548

In [73]:
# For the selected season
data = matches.loc[matches['season'] == 2012]
data = deliveries[deliveries['match_id'].isin(data.id)]
data.groupby(['batsman'])['batsman_runs'].sum().reset_index().sort_values(by='batsman_runs', ascending=False)[:1]

batsman  batsman_runs
36  CH Gayle           733

### Most number of catches by a fielder in a match for the selected season.

In [74]:
deliveries.columns

Index(['match_id', 'inning', 'batting_team', 'bowling_team', 'over', 'ball',
       'batsman', 'non_striker', 'bowler', 'is_super_over', 'wide_runs',
       'bye_runs', 'legbye_runs', 'noball_runs', 'penalty_runs',
       'batsman_runs', 'extra_runs', 'total_runs', 'player_dismissed',
       'dismissal_kind', 'fielder'],
      dtype='object')

In [84]:
data = matches.loc[matches['season'] == 2012]
data = deliveries[deliveries['match_id'].isin(data.id)]
data[data['dismissal_kind'] == 'caught'].groupby(['fielder'])['dismissal_kind'].count().sort_values(ascending=False)[:5]

fielder
NV Ojha      14
RG Sharma    13
MS Dhoni     12
N Saini      11
AM Rahane    11
Name: dismissal_kind, dtype: int64